common/fs/src/cache/mod.rs

```diff
@@ -35,6 +35,7 @@ type WatchDescriptors = HashMap<WatchDescriptor, Vec<EntryKey>>;
 pub type EntryKey = DefaultKey;
 
 type EntryMap<T> = SlotMap<EntryKey, RefCell<entry::Entry<T>>>;
+type FsResult<T> = Result<T, String>;
```

Comment:
By having this as a String it flattens a few different kinds of errors into one(String), so we lose some information if we wanted to handle it higher up.

Could we add an FsCacheError enum and make this type FsResult<T> = Result<T, FsCacheError> ?

The thiserror crate makes creating these kinds of strongly typed errors very easy

Reply:
For examples of different kinds of errors I think there are Watch, WatchEvent, CacheLookup , InvalidFsObject at least, there may be more in there. It would be useful to be able to tell them apart

Reply:
Makes sense, I'll change it to use an enum.

Reply:
I've added an enum to represent the different errors.